### PR TITLE
AKS Rancher HA automation

### PR DESCRIPTION
**Pull request by izaac**
Using Terraform to manage the Group and AKS resources.
K8s version, location and other variables can be set using environment variables.
Task: https://github.com/rancher/rancher/issues/30957

### CODE REVIEW

File: tests/validation/tests/v3_api/test_create_ha.py [hunk 1]
```diff
@@ -244,6 +255,24 @@ def set_route53_with_eks_ingress():
     time.sleep(60)
 
 
+def set_route53_with_aks_ingress():
```
Comment (jiaqiluo) [inline tests/validation/tests/v3_api/test_create_ha.py:258]: I am confused here: Why do you use AWS's route53 for an AKS cluster?

**Review by jiaqiluo — CHANGES_REQUESTED** [review]

File: tests/validation/tests/v3_api/test_create_ha.py (hunk 1 above)
Comment (izaac) [reply tests/validation/tests/v3_api/test_create_ha.py:258]: DNS is platform agnostic, and we have the logic to manage route53 already. 
Second reason the DNS in Azure is global and I don't want to touch it if possible.

File: tests/validation/tests/v3_api/Jenkinsfile [hunk 2]
```diff
@@ -69,7 +69,11 @@ node {
                         string(credentialsId: 'RANCHER_FREEIPA_GROUP_SEARCH_BASE', variable: 'RANCHER_FREEIPA_GROUP_SEARCH_BASE'),
                         string(credentialsId: 'RANCHER_FREEIPA_AUTH_USER_PASSWORD', variable: 'RANCHER_FREEIPA_AUTH_USER_PASSWORD'),
                         string(credentialsId: 'RANCHER_FREEIPA_HOSTNAME_OR_IP_ADDRESS', variable: 'RANCHER_FREEIPA_HOSTNAME_OR_IP_ADDRESS'),
-                        string(credentialsId: 'RANCHER_FREEIPA_SPECIAL_CHAR_PASSWORD', variable: 'RANCHER_FREEIPA_SPECIAL_CHAR_PASSWORD')
+                        string(credentialsId: 'RANCHER_FREEIPA_SPECIAL_CHAR_PASSWORD', variable: 'RANCHER_FREEIPA_SPECIAL_CHAR_PASSWORD'),
+                        string(credentialsId: 'ARM_SUBSCRIPTION_ID', variable: 'ARM_SUBSCRIPTION_ID'),
```
Comment (jiaqiluo) [inline tests/validation/tests/v3_api/Jenkinsfile:73]: What does `ARM` stand for?

File: tests/validation/tests/v3_api/test_create_ha.py (hunk 1 above)
Comment (jiaqiluo) [reply tests/validation/tests/v3_api/test_create_ha.py:258]: In this method it calls `AmazonWebServices().upsert_route_53_record_cname` which is from the AWS module. How does it work with AKS?

**Review by jiaqiluo — CHANGES_REQUESTED** [review]

File: tests/validation/tests/v3_api/Jenkinsfile (hunk 2 above)
Comment (izaac) [reply tests/validation/tests/v3_api/Jenkinsfile:73]: That's for Azure ARM provider (Terraform) https://github.com/terraform-providers/terraform-provider-azurerm

**Comment by bmdepesa** [comment]
Can we add code to delete the AKS cluster by name when it's no longer needed? (this would need a job definition too)

File: tests/validation/Dockerfile.v3api [hunk 3]
```diff
@@ -39,5 +39,6 @@ RUN wget https://storage.googleapis.com/kubernetes-release/release/$KUBECTL_VERS
     wget https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip && \
     unzip awscli-exe-linux-x86_64.zip && \
     ./aws/install && \
+    curl -sL https://aka.ms/InstallAzureCLIDeb | bash && \
```
Comment (bmdepesa) [inline tests/validation/Dockerfile.v3api:42]: Where is this used?

**Review by bmdepesa — CHANGES_REQUESTED** [review]

File: tests/validation/Dockerfile.v3api (hunk 3 above)
Comment (izaac) [reply tests/validation/Dockerfile.v3api:42]: azure cli is used by terraform.

**Comment by izaac** [comment]
> Can we add code to delete the AKS cluster by name when it's no longer needed? (this would need a job definition too)

@bmdepesa Im planning on the delete for a separate PR

File: tests/validation/tests/v3_api/Jenkinsfile (hunk 2 above)
Comment (izaac) [reply tests/validation/tests/v3_api/Jenkinsfile:73]: Stands for Azure Resource Manager

**Review by bmdepesa — APPROVED** [review]

**Review by jiaqiluo — APPROVED** [review]
LGTM

File: tests/validation/tests/v3_api/test_create_ha.py [hunk 4]
```diff
@@ -404,6 +433,34 @@ def create_rke_cluster_config(aws_nodes):
     return clusterfilepath
 
 
+def create_aks_cluster():
+    tf_dir = DATA_SUBDIR + "/" + "terraform/aks"
+    aks_k8_s_version = os.environ.get('RANCHER_AKS_K8S_VERSION', '')
+    aks_location = os.environ.get('RANCHER_AKS_LOCATION', '')
+    client_id = os.environ.get('ARM_CLIENT_ID', '')
+    client_secret = os.environ.get('ARM_CLIENT_SECRET', '')
```
Comment (sowmyav27) [inline tests/validation/tests/v3_api/test_create_ha.py:441]: We don't need tenant ID and Subscription ID here to create the cluster?

Comment (izaac) [reply tests/validation/tests/v3_api/test_create_ha.py:441]: That's loaded from the ARM variables and used by the azure cli client.
I use these two to pass these values to the cluster creation https://github.com/rancher/rancher/pull/30987/files#diff-eb90b3de3553e328d8a4b6c4e0fbbf42db2a2612ef538c5bf9399dfd2ae8c883R37

**Review by sowmyav27 — APPROVED** [review]
LGTM

**Pull request merged**